src/Controls/StreetControls.js

```diff
@@ -0,0 +1,111 @@
+import FirstPersonControls from 'Controls/FirstPersonControls';
+import AnimationPlayer, { AnimatedExpression } from 'Core/AnimationPlayer';
+
+// Expression used to damp camera's moves
+function moveCameraExp(root, progress) {
+    const dampingProgress = 1 - Math.pow((1 - (Math.sin((progress - 0.5) * Math.PI) * 0.5 + 0.5)), 2);
+    root.camera.position.lerpVectors(root.positionFrom, root.positionTo, dampingProgress);
+}
+/**
+ * @classdesc Camera controls that can follow a path.
+ * It is used to simulate a street view.
+ * It stores a currentPosition and nextPosition, and do a camera traveling to go to next position.
+ *
+ * <ul> Bindings inherited from FirstPersonControls
+ * <li><b> up + down keys : </b> forward/backward </li>
+ * <li><b> left + right keys: </b> strafing movements </li>
+ * <li><b> pageUp + pageDown: </b> vertical movements </li>
+ * <li><b> mouse click+drag: </b> pitch and yaw movements (as looking at a panorama) </li>
+ * </ul>
+ * <ul> Bindings added
+ * <li><b> keys Z : </b> Move camera to the next position </li>
+ * </ul>
+ * @extends FirstPersonControls
+ */
+class ImmersiveControls extends FirstPersonControls {
+    /**
+     * @constructor
+     * @param { View } view - View where this control will be used
+     * @param { Object } options - Configuration of this controls
+     * @param { number } options.animationDuration - Duration of the animation to another panoramic.
+     */
+    constructor(view, options = {}) {
+        super(view, options);
+
+        this.isStreetControls = true;
+
+        // store layers
+        this.layers = [];
+        this.currentLayerIndex = 0;
+
+        // manage camera movements
+        this.player = new AnimationPlayer();
+        this.animationMoveCamera = new AnimatedExpression({
+            duration: options.animationDuration || 50,
+            root: this,
+            expression: moveCameraExp,
+            name: 'Move camera',
+        });
+        this.player.addEventListener('animation-frame', this.updateView.bind(this));
+
+        // twos positions used by this control : current and next
+        this.currentPosition = undefined;
+        this.nextPosition = undefined;
+    }
+
+    setCurrentPosition(newCurrentPosition) {
+        this.currentPosition = newCurrentPosition;
+    }
+
+    setNextPosition(newNextPosition) {
+        this.nextPosition = newNextPosition;
+    }
+
+    /**
+     * Sets the camera to the current position, looking at the next position.
+     */
+    setCameraToCurrentPosition() {
+        this.setCameraOnPosition(this.currentPosition, this.nextPosition);
+    }
+
+    /**
+     * Set the camera on a position, looking at another position.
+     *
+     * @param      { THREE.Vector3 }  currentPosition   The position to set the camera
+     * @param      { THREE.Vector3 }  nextPosition      The position where the camera look at.
+     */
+    setCameraOnPosition(currentPosition, nextPosition) {
```

Comment:
`currentPosition` could you rename in `position` it's more generic and `nextPosition` to `lookAt` it's more adapted.

Reply:
yep, good idea